### PR TITLE
Add controller agent configuration watcher worker as a dependency of db-accessor

### PR DESCRIPTION
**Pull request by manadart**
This is boilerplate in preparation for removing the `peergrouper` dependency from the `dbaccessor` worker.

The worker now depends on the `controlleragentconfig` worker's `ConfigWatcher` output.

In future patches we will get notified of controller configuration changes, whereupon we will process any configured cluster members.

## QA steps

No functional changes at this point. Bootstrap to test for regression.

## Links

**Jira card:** [JUJU-5308](https://warthogs.atlassian.net/browse/JUJU-5308)

[JUJU-5308]: https://warthogs.atlassian.net/browse/JUJU-5308?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: internal/worker/dbaccessor/worker_integration_test.go [hunk 1]
```diff
@@ -291,3 +292,17 @@ func (r *txnRunner) Txn(ctx context.Context, f func(context.Context, *sqlair.TX)
 func (r *txnRunner) StdTxn(ctx context.Context, f func(context.Context, *sql.Tx) error) error {
 	return errors.Trace(database.StdTxn(ctx, r.db, f))
 }
+
+type controllerConfigWatcher struct{}
```
Comment (manadart) [inline internal/worker/dbaccessor/worker_integration_test.go:296]: These integration tests are for testing real database access, so this is not needed at present aside from supplying a non-nil dependency. I kept the convention of eschewing mocks in these tests.

**Review by SimonRichardson — APPROVED** [review]

**Pull request merged**